### PR TITLE
GCI: Add test for delete_workers in aws_utils.

### PR DESCRIPTION
**Pull request by abinav62**

### CODE REVIEW

File: tests/unit/challenges/test_aws_utils.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+            self.ecs_client,
+            self.challenge3,
+            self.client_token)
+
+        pk_list = [self.challenge.pk, self.challenge2.pk, self.challenge3.pk]
+        queryset = super(TestDeleteWorkers, self).queryset(pk_list)
+
+        count = 3
+        expected_workers = [None, None, None]
+        failures = []
```
Comment (KhalidRmb) [inline tests/unit/challenges/test_aws_utils.py]: Please change the variable name from `failures` to `expected_failures`.  That would be more clear.

File: tests/unit/challenges/test_aws_utils.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        expected_response = {"count": count, "failures": failures}
```
Comment (KhalidRmb) [inline tests/unit/challenges/test_aws_utils.py]: Same.

File: tests/unit/challenges/test_aws_utils.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+            self.client_token)
+
+        pk_list = [self.challenge.pk, self.challenge2.pk, self.challenge3.pk]
+        queryset = super(TestDeleteWorkers, self).queryset(pk_list)
+
+        message = "Please select challenges with active workers only."
+
+        count = 2
+        expected_workers = [None, None, None]
+        failures = [{"message": message, "challenge_pk": self.challenge.pk}]
```
Comment (KhalidRmb) [inline tests/unit/challenges/test_aws_utils.py]: Change name to `expected_failures`.

File: tests/unit/challenges/test_aws_utils.py [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+        pk_list = [self.challenge.pk, self.challenge2.pk, self.challenge3.pk]
+        queryset = super(TestDeleteWorkers, self).queryset(pk_list)
+
+        message = "errMessage"
+        exception_message = {"Error": message, "ResponseMetadata": {"HTTPStatusCode": HTTPStatus.NOT_FOUND}}
+        mock_delete_service_by_pk.return_value = exception_message
+
+        count = 0
+        failures = [
```
Comment (KhalidRmb) [inline tests/unit/challenges/test_aws_utils.py]: Same.

File: tests/unit/challenges/test_aws_utils.py [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+            self.client_token)
+        aws_utils.create_service_by_challenge_pk(
+            self.ecs_client,
+            self.challenge3,
+            self.client_token)
+
+        pk_list = [self.challenge.pk, self.challenge2.pk, self.challenge3.pk]
+        queryset = super(TestDeleteWorkers, self).queryset(pk_list)
+
+        count = 3
```
Comment (KhalidRmb) [inline tests/unit/challenges/test_aws_utils.py]: Change from `count` to `expected_count`. Do the same everywhere.

File: tests/unit/challenges/test_aws_utils.py [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+        pk_list = [self.challenge.pk, self.challenge2.pk, self.challenge3.pk]
+        queryset = super(TestDeleteWorkers, self).queryset(pk_list)
+
+        message = "errMessage"
+        exception_message = {"Error": message, "ResponseMetadata": {"HTTPStatusCode": HTTPStatus.NOT_FOUND}}
+        mock_delete_service_by_pk.return_value = exception_message
+
+        count = 0
+        failures = [
+            {"message": message, "challenge_pk": self.challenge.pk},
```
Comment (KhalidRmb) [inline tests/unit/challenges/test_aws_utils.py]: Create a new variable called 'expected_message` with the same value as `message` and use that here.  For the sake of clarity. Do the same in other places.

**Comment by KhalidRmb** [comment]
@abinav62 Also, why are you removing the deploymentController key from the dictionary?

**Comment by abinav62** [comment]
I had an error I couldn't solve relating to the deploymentController. If I delete it, it works, but while its there it creates a parameter invalid error.

**Comment by abinav62** [comment]
I could not come up with a better fix for it either.

**Comment by abinav62** [comment]
See https://github.com/Cloud-CV/EvalAI/pull/2600/files. They also did the same thing.

**Comment by takitsuse** [comment]
I made that change in reference to https://github.com/Cloud-CV/EvalAI/pull/2596/files, FYI.

**Comment by abinav62** [comment]
I know, but I tried it because I was having an error with the deployment controller. If I keep it, it creates a parameter invalid error, and I couldn't come up with a better fix.

**Comment by KhalidRmb** [comment]
Hey @abinav62. This seems to be a weird issue since everything's is supposed to be working fine without the modification. I'll look into this a bit more and get back for further review. 

Please continue work on other tasks if you want to.

**Comment by abinav62** [comment]
Should I abandon the task? Would abandoning the task reduce my chances for being chosen as a finalist?

**Comment by abinav62** [comment]
I would like to claim another task, but I can't do so unless the task is approved or I abandon.

**Comment by KhalidRmb** [comment]
Hey, you can pick it up later. It won't hurt you :)

**Comment by KhalidRmb** [comment]
You can make the changes as in #2604 for the deploymentController.

**Comment by abinav62** [comment]
@yashdusing @KhalidRmb Thanks for the help!

File: tests/unit/challenges/test_aws_utils.py [hunk 7]
```diff
@@ -0,0 +1,214 @@
+import boto3
+import mock
+import os
+
+from datetime import timedelta
+from http import HTTPStatus
+from moto import mock_ecs
+
+from allauth.account.models import EmailAddress
+from django.contrib.auth.models import User
```
Comment (Ram81) [inline tests/unit/challenges/test_aws_utils.py]: @abinav62 can you reorder imports lexicographically

**Comment by abinav62** [comment]
@Ram81 PTAL.

File: tests/unit/challenges/test_aws_utils.py [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+        expected_count = 0
+        expected_failures = [
+            {"message": expected_message, "challenge_pk": self.challenge.pk},
+            {"message": expected_message, "challenge_pk": self.challenge2.pk},
+            {"message": expected_message, "challenge_pk": self.challenge3.pk}
+        ]
+        expected_response = {"count": expected_count, "failures": expected_failures}
+        response = aws_utils.delete_workers(queryset)
+        assert expected_response == response
+        assert list(c.workers for c in queryset) == [1, 1, 1]
```
Comment (yashdusing) [inline tests/unit/challenges/test_aws_utils.py]: @abinav62 could you store `[1, 1, 1]` as `expected_workers` ?

**Comment by abinav62** [comment]
@yashdusing PTAL.

**Comment by yashdusing** [comment]
LGTM 👍

**Pull request closed without merging**